### PR TITLE
Store list of kept packages and report the number of them in motd

### PR DESCRIPTION
**Pull request by rbalint**
LP: #1823070

### CODE REVIEW

File: unattended-upgrade [hunk 1]
```diff
@@ -87,6 +87,7 @@ import distro_info
 
 # the reboot required flag file used by packages
 REBOOT_REQUIRED_FILE = "/var/run/reboot-required"
+KEPT_PACKAGES_FILE = "var/lib/unattended-upgrades/kept-back"
```
Comment (tdaitx) [inline unattended-upgrade]: Isn't it missing a leading / ?

Comment (rbalint) [reply unattended-upgrade]: No, I'm transitioning to prepend rootdir to global paths an it marks that.

**Comment by julian-klode** [comment]
I'd really love to see a complete motd example, to see how this fits in with available updates messaging, especially concerning ESM (I think I can figure that out from a complete example on eoan)

**Comment by rbalint** [comment]
@julian-klode This is how it looks like on Bionic with no ESM updates:
```
Welcome to Ubuntu 18.04.2 LTS (GNU/Linux 4.18.0-21-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Tue Jun 11 12:51:34 UTC 2019

  System load:    1.38      Processes:           24
  Usage of /home: unknown   Users logged in:     0
  Memory usage:   0%        IP address for eth0: 10.84.73.183
  Swap usage:     31%

 * Ubuntu's Kubernetes 1.14 distributions can bypass Docker and use containerd
   directly, see https://bit.ly/ubuntu-containerd or try it now with

     snap install microk8s --classic

  Get cloud support with Ubuntu Advantage Cloud Guest:
    http://www.ubuntu.com/business/services/cloud

 * Canonical Livepatch is available for installation.
   - Reduce system reboots and improve kernel security. Activate at:
     https://ubuntu.com/livepatch

19 packages can be updated.
8 updates are security updates.


2 packages could not be upgraded by unattended-upgrades
  - Please see /var/log/unattended-upgrades/unattended-upgrades.log
    for more information.

*** System restart required ***
```
I think the protocol is leaving on blank line _before_ each item, but update-notifier prints a blank line _after_ the its message, too, which looks bad:

```
root@bb-qemu:~/unattended-upgrades# ls /etc/update-motd.d/90-updates-available 
/etc/update-motd.d/90-updates-available
root@bb-qemu:~/unattended-upgrades# cat /etc/update-motd.d/90-updates-available 
#!/bin/sh

stamp="/var/lib/update-notifier/updates-available"

[ ! -r "$stamp" ] || cat "$stamp"
root@bb-qemu:~/unattended-upgrades# cat /var/lib/update-notifier/updates-available

19 packages can be updated.
8 updates are security updates.

root@bb-qemu:~/unattended-upgrades# dpkg -S /etc/update-motd.d/90-updates-available
update-notifier-common: /etc/update-motd.d/90-updates-available
```

Maybe the unattended-upgrades script could skip the blank line _before_ its text if `/var/lib/update-notifier/updates-available` exists thus the output would look like:

```
...

19 packages can be updated.
8 updates are security updates.
2 packages could not be upgraded by unattended-upgrades
  - Please see /var/log/unattended-upgrades/unattended-upgrades.log
    for more information.

...
```

**Comment by julian-klode** [comment]
This basically looks OK, but we should try to get the paragraph down to 2 lines such as

```
2 updates could not be installed automatically.
See /var/log/unattended-upgrades/unattended-upgrades.log for more details.
```

or

```
2 packages could not be updated automatically.
See /var/log/unattended-upgrades/unattended-upgrades.log for more details.
```

**Comment by julian-klode** [comment]
Or really evil:

```
2 updates could not be installed automatically. For more details,
see /var/log/unattended-upgrades/unattended-upgrades.log
```

**Comment by julian-klode** [comment]
That fits in better after the ESM messaging I believe:

```
15 updates can be installed immediately.
7 of these updates are security updates.

Enable ESM to receive 23 additional security updates.
See 'ua enable esm' or https://ubuntu.com/esm
```

**Comment by rbalint** [comment]
@julian-klode I just fixed the format

**Comment by julian-klode** [comment]
> 2 packages could not be installed automatically

um, this looks wrong, either you wanted "updates ... installed ..." or "packages ... updated ..."

**Comment by julian-klode** [comment]
LGTM!

**Pull request merged**